Sanguine Soul/sanguinesoul.asl

```diff
@@ -54,36 +57,41 @@ init
 update
 {
     // Deref useful FNames here
-    IDictionary<string, object> currdict = current;
-    foreach (var fname in new List<string>(currdict.Keys))
+    IDictionary<string, object> currentLookup = current;
+    foreach (var entry in currentLookup)
     {
-        if (!fname.EndsWith("FName"))
+	string key = entry.Key;
+	object value = entry.Value
+
+        if (!key.EndsWith("FName"))
             continue;
         
         // e.g. missionFName -> mission
-        var key = fname.Substring(0, fname.Length-5);
-        var val = vars.ReadFName((long)currdict[fname]);
-        if (val == "None" && currdict.ContainsKey(key))
+        key = key.Substring(0, fname.Length - 5);
+
+	string fNameEntry = vars.ReadFName(value);
+        if (fNameEntry == "None")
```

Comment:
This needs to check if the key is already present in `current`. We want to skip setting it to None only if there is no other value already present.

Reply:
I don't follow here. Could you explain?

Do you want to set the value to "None" only exactly once?

Reply:
If we don't set it on the first iteration, then it will be null wherever else we try to check it. This will most likely cause issues.

From what I recall, the specific use case for this ASL is that the world's name briefly goes to None when it changes. It should start as None if it's the first read of it, but once it finds a value it should hold onto it until it changes to another, valid value.

Reply:
That makes sense.

Reply:
Given all other comments on this are resolved, merging this PR should just need this issue to be fixed